src/components/Project/UploadButton.tsx

```diff
@@ -2,44 +2,79 @@
 
 import { uploadFile } from '@/lib/ipfs/uploadToIpfs'
 import { Button } from '../ui/button'
-import { FilePlusIcon } from '@radix-ui/react-icons'
+import { FilePlusIcon, ReloadIcon } from '@radix-ui/react-icons'
 import usePaymasterAttest from '@/hooks/project/usePaymasterAttest'
 import { useProjectProvider } from '@/context/ProjectProvider'
-import { useState } from 'react'
+import { useRef, useState } from 'react'
+import { toast } from '../ui/use-toast'
 
 const UploadButton = () => {
-  const { setAnimationUrl } = useProjectProvider()
+  const {
+    setAnimationUrl,
+    uploadingAudio,
+    setUploadingAudio,
```

Comment:
same question here.
- is it essential to store this state in the provider?
- `<UploadButton />` the only file using these variables?